github/pkg/reconciler/githubsource.go

```diff
@@ -186,8 +187,9 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, source *sourcesv1alpha1.G
 		// Delete the webhook using the access token and stored webhook ID
 		err = r.deleteWebhook(ctx, args)
 		if err != nil {
```

Comment:
same here.

Reply:
I have used errors.As() for this. Please review next commit. Do you think that is beautiful code?